### DISCUSSION TITLE
Reading the file using chain command

### DISCUSSION THREAD

**Discussion by Naranthiran** [ipxe/ipxe#1366] [General]
Hi,

I have a legacy systems, were I have files ipxe.iso(this iso is enbeded with the script to read the file chain file:/config.ipxe) file mounted and copied to the directory /boot/iso/

 ‘/boot/iso/boot.catalog’
 ‘/boot/iso/esp.img’
 ‘/boot/iso/ipxe.lkrn’
 ‘/boot/iso/isolinux.bin’
 ‘/boot/iso/isolinux.cfg’
 ‘/boot/iso/ldlinux.c32’
I also have a file call config.ipxe in the same directory.
 ‘/boot/iso/config.ipxe’

This works well on UEFI systems but does not work on legacy systems. Does legacy BISO support file reading? or can any one suggest the correct approch?

Thanks in advance

Regards
Naranthiran Duraisamy

**Comment by NiKiZe** [comment]
This is not a bug or something that is likely to change, so moving from issue to discussion.

Bios has no concept of filesystem, iPXE does support file access only thru the efi firmware support.

However in pcbios mode there is support for providing data to ipxe thru initrd, so if you are using standard isolinux to start ipxe.lkrn then the file could be supplied that way. This is also supported in a generic way, but then often with the filename autoexec.ipxe.

**Comment by Naranthiran** [comment]
Hi @NiKiZe ,

Thanks for your reply!

I am not using Initrd, I will load the kernel(/boot/iso/ipxe.lkrn) which the embedded with code(chain file:/config.ipxe) and the config.ipxe contains the details of network setup and from where initrd and vmlinuz files are to be loaded.

So with this method, I am had a confusion whether I am placing the files(config.ipxe) in the right path and whether legacy supports file reading..

Is there any other way to read the file with the approach I am trying other than ipxe thru initrd?
